File: content-solutions/sorting/leaderboard.js
function solution(names, scores, times) {
  const order = names.map((_, i) => i);
  order.sort(
    (a, b) => scores[b] - scores[a] || times[a] - times[b] || (names[a] < names[b] ? -1 : names[a] > names[b] ? 1 : 0),
  );
  return order.map((i) => names[i]);
}

File: content-solutions/sorting/leaderboard.java
import java.util.*;

class Solution {
    public String[] solution(String[] names, int[] scores, int[] times) {
        Integer[] order = new Integer[names.length];
        for (int i = 0; i < order.length; i++) order[i] = i;
        Arrays.sort(order, Comparator
            .comparingInt((Integer i) -> -scores[i])
            .thenComparingInt(i -> times[i])
            .thenComparing(i -> names[i]));
        String[] answer = new String[names.length];
        for (int i = 0; i < order.length; i++) answer[i] = names[order[i]];
        return answer;
    }
}
